### DISCUSSION TITLE
I got error when I setup RBD mirror daemon, keyring file is not found

### DISCUSSION THREAD

**Discussion by ye0321** [rook/rook#9459] [Q&A]
I try to setup RBDmirror followed by: https://rook.github.io/docs/rook/v1.7/rbd-mirroring.html, when executed:
kubectl create -f <ROOK_CEPH_PATH>/rbdmirror.yaml and create daemon pod, i got below error when i looked into pod's log

```
debug 2021-12-17T11:55:42.395+0000 7efda50e45c0 -1 auth: unable to find a keyring on /etc/ceph/c214a7a1-6295-48cc-9cb0-272c50d1724f.client.admin.keyring,/etc/ceph/c214a7a1-6295-48cc-9cb0-272c50d1724f.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
```

i try to look into pod, i actually not find above path but below is over there:
```
[root@rook-ceph-rbd-mirror-a-799768db44-5v5xv keyring-store]# cat /etc/ceph/keyring-store/keyring 

[client.rbd-mirror.a]
        key = AQDGX7xhxeNANRAA4oNtM+3qrENlsxEiauRYRg==
        caps mon = "profile rbd-mirror"
        caps osd = "profile rbd"
[root@rook-ceph-rbd-mirror-a-799768db44-5v5xv keyring-store]# 
```

I am not sure why the path of keyring file is wrong, how do I change it to correct path?

**Comment by leseb** [comment]
This error is a bit misleading and does not necessarily mean there is a connection issue. Have you tried setting up the mirroring further? Or is the daemon crashing? Thanks

**Comment by ye0321** [comment]
Hi leseb,

thank you for response at first. I check daemon 's status, actually, it is 'error';
I looked into the log of daemon's pod, it is:
debug 2021-12-23T13:21:40.840+0000 7fb1fe435540 0 rbd::mirror::PoolReplayer: 0x561b12175800 init_rados: reverting global config option override: mon_host: [v2:10.97.35.147:3300,v1:10.97.35.147:6789],[v2:10.109.149.149:3300,v1:10.109.149.149:6789],[v2:10.108.190.162:3300,v1:10.108.190.162:6789] ->
debug 2021-12-23T13:21:40.840+0000 7fb1fe435540 -1 Errors while parsing config file!
debug 2021-12-23T13:21:40.840+0000 7fb1fe435540 -1 can't open 513ce4b5-87e7-498d-bbee-278d5c6c86a0.conf: (2) No such file or directory
debug 2021-12-23T13:21:40.840+0000 7fb1fe435540 -1 auth: unable to find a keyring on /etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.client.rbd-mirror.a.keyring,/etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
debug 2021-12-23T13:21:40.841+0000 7fb1fe435540 -1 auth: unable to find a keyring on /etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.client.rbd-mirror.a.keyring,/etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
debug 2021-12-23T13:21:40.841+0000 7fb1fe435540 -1 auth: unable to find a keyring on /etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.client.rbd-mirror.a.keyring,/etc/ceph/513ce4b5-87e7-498d-bbee-278d5c6c86a0.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
debug 2021-12-23T13:26:40.841+0000 7fb1fe435540 0 monclient(hunting): authenticate timed out after 300

'513ce4b5-87e7-498d-bbee-278d5c6c86a0' is peer cluster id. 
actually I build 2 cluster in 2 different VPCs, the subnet is different and can not connect directly. I use NAT to forward port to make connection, so i am not sure if is is valid case for ceph RDB mirror.

in other word, 2 k8s cluster's pod can not connect directly,  i have to make connection by router in public level. so mon's address is private and cannot be connected by peer cluster, i use below command to set peer manully:

rbd --cluster ceph mirror pool peer add mirrored-pool client.rbd-mirror-peer@6ac1a0da-2d66-4a40-aac3-c58bb9ed8229 --remote-mon-host 10.39.74.23:30441,10.39.74.23:31360,10.39.74.23:32050 --remote-key-file remote-key-file

actully, in ceph.conf, the mon's host like: 10.250.0.16:3300 ...

**Reply by leseb** [reply]
If you have NAT in place you need to advertise the address of the NAT and not the background IP which I believe in your case is `10.250.0.16`. The `rbd` CLI report the mon IP as it knows it from its current ceph.conf.
All that said, I'm not sure if the mons will accept connections from a NAT...
